lib/src/hash_sink.dart

```diff
@@ -44,6 +44,9 @@ abstract class HashSink implements Sink<List<int>> {
   /// This should be updated each time [updateHash] is called.
   Uint32List get digest;
 
+  /// The length
```

Comment:
See next comment first! (And if you follow that, this might be moot).

The length of what? What does it mean, where is it used?
What is a "signature" (first mention in this class/file)?

Maybe:
> /// The number of signature bytes emitted at the end of the message.
> /// 
> /// An encrypted message is followed by a signature which depends
> /// on the encryption algorithm used. This value specifies the 
> /// number of bytes used by this signature. It must always be
> /// a power of 2 and no less than 8.

(If that is actually correct).

Comments should be a full statement, or at least a noun phrase, with a full stop at the end.

Should the class documentation say that subclasses should/can override `signatureBytes`?